From 05c8c4b704c2135a50b1546400c617850263670a Mon Sep 17 00:00:00 2001
From: Frantisek Mikulu <frantisek.mikulu@openhs.org>
Date: Sat, 7 Jan 2017 10:50:37 +0100
Subject: [PATCH] Fixed build with SSL

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3960a8f..e8aadb8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -88,6 +88,7 @@ IF (PAHO_WITH_SSL)
         paho-mqtt3cs paho-mqtt3as PROPERTIES
         VERSION ${CLIENT_VERSION}
         SOVERSION ${PAHO_VERSION_MAJOR})
+        COMPILE_FLAGS -DOPENSSL)
     INSTALL(TARGETS paho-mqtt3cs
         RUNTIME DESTINATION bin
         ARCHIVE DESTINATION lib
-- 
1.9.5.msysgit.0

